samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,241 @@
+        try (var propertiesFileInputStream = new FileInputStream(fileName)) {
+            Properties prop = new Properties();
+            prop.load(propertiesFileInputStream);
+            return prop;
+        }
+    }
+
+    /**
+     * Run all sample steps in one single test.
+     * Note: Sample steps cannot be separated into single tests because {@link EdcRuntimeExtension}
+     * runs before each single test.
+     */
+    @Test
+    void runSampleSteps() throws IOException {
+        assertTestPrerequisites();
+
+        assertInitiateContractNegotiation();
+        assertLookUpContractAgreementId();
+        assertRequestFile();
+        assertWaitForDestinationFileExistence();
+
+        cleanTemporaryTestFiles();
+    }
+
+    /**
+     * Assert that prerequisites are fulfilled before running the test.
+     * This assertion checks only whether the file to be copied is not existing already.
+     */
+    void assertTestPrerequisites() {
+        assertThat(destinationFile).doesNotExist();
+    }
+
+    /**
+     * Remove files created while running the tests.
+     * The copied file will be deleted.
+     */
+    void cleanTemporaryTestFiles() {
+        destinationFile.delete();
+    }
+
+    /**
+     * Assert that the file to be copied exists at the expected location.
+     * This method waits a duration which is defined in {@link FileTransferSampleTest#TIMEOUT}.
+     */
+    void assertWaitForDestinationFileExistence() {
+        await().atMost(TIMEOUT).pollInterval(POLL_INTERVAL).untilAsserted(() -> assertThat(destinationFile).exists());
+    }
+
+    /**
+     * Assert that a POST request to initiate a contract negotiation is successful.
+     * This method corresponds to the command in the sample: curl -X POST -H "Content-Type: application/json" -H "X-Api-Key: password" -d @samples/04.0-file-transfer/contractoffer.json "http://localhost:9192/api/v1/data/contractnegotiations"
+     */
+    void assertInitiateContractNegotiation() {
+        JsonPath jsonPath = RestAssured
+            .given()
+                .headers(API_KEY_HEADER_KEY, API_KEY_HEADER_VALUE)
+                .contentType(ContentType.JSON)
+                .body(new File(TestUtils.findBuildRoot(), CONTRACT_OFFER_FILE_PATH))
+            .when()
+                .post(INITIATE_CONTRACT_NEGOTIATION_URI)
+            .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body("id", not(emptyString()))
+                .extract()
+                .jsonPath();
+
+        contractNegotiationId = jsonPath.get("id");
+    }
+
+    /**
+     * Assert that a GET request to look up a contract agreement is successful.
+     * This method corresponds to the command in the sample: curl -X GET -H 'X-Api-Key: password' "http://localhost:9192/api/v1/data/contractnegotiations/{UUID}"
+     */
+    void assertLookUpContractAgreementId() {
+        var localContractAgreementId = new AtomicReference<String>();
```

Comment:
why is this needed, instead of directly writing into the field?

Reply:
As it turned out, it's not necessary. Changed it.

Pattern was taken from `java/org/eclipse/dataspaceconnector/test/e2e/Participant.java`.